extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -327,6 +339,34 @@ private static void contributeMethodsWithInheritedValidation(Map<DotName, Set<St
         }
     }
 
+    private static Map<DotName, Set<String>> gatherJaxRsMethods(
+            List<AdditionalJaxRsResourceMethodAnnotationsBuildItem> additionalJaxRsResourceMethodAnnotations,
+            IndexView indexView) {
+        Map<DotName, Set<String>> jaxRsMethods = new HashMap<>();
+
+        Collection<DotName> effectiveJaxRsMethodDefiningAnnotations = new ArrayList<>(
+                JAXRS_METHOD_ANNOTATIONS.length + additionalJaxRsResourceMethodAnnotations.size());
+        effectiveJaxRsMethodDefiningAnnotations.addAll(Arrays.asList(JAXRS_METHOD_ANNOTATIONS));
```

Comment:
I think it would be better conceptually to pass the full list to the method rather than building it inside.